src/dll.py

```diff
@@ -0,0 +1,110 @@
+"""Write an implementation of linked lists in Python."""
+
+
+class Node(object):
+    """Node class for use in double linked list."""
+
+    def __init__(self, data, next_node=None, prior_node=None):
+        """Initialize a node when adding to the linked list."""
+        self.data = data
+        self.next_node = next_node
+        self.prior_node = prior_node
+
+
+class DoubleLinkedList(object):
+    """Double linked version of a list."""
+
+    def __init__(self):
+        """Initialize an empty double linked list."""
+        self._length = 0
+        self.tail = None
+        self.head = None
+
+    def push(self, val):
+        """Push a value to the head of the list."""
+        if not val:
+            raise TypeError('Please provide a not null value.')
+        self._length += 1
```

Comment:
It might be best if you put this at the end of the function, after the node is actually added to your list.